ex_app/lib/tools.py

```diff
@@ -40,5 +49,33 @@ def get_tools(nc: Nextcloud):
 				print(f"{function_name} in {module_name} is not callable.")
 		else:
 			print(f"{function_name} not found in {module_name}.")
+	get_categories()
 
 	return safe_tools, dangerous_tools
+
+def get_categories():
+	directory = dirname(__file__) + '/all_tools'
+	function_name = "get_category_name"
+
+	categories = {}
+
+	py_files = [f for f in os.listdir(directory) if f.endswith(".py") and f != "__init__.py"]
+
+	for file in py_files:
+		module_name = pathlib.Path(file).stem  # Extract module name without .py
+		module_path = os.path.join(directory, file)
+
+		# Load module dynamically
+		spec = importlib.util.spec_from_file_location(module_name, module_path)
+		module = importlib.util.module_from_spec(spec)
+		spec.loader.exec_module(module)
```

Comment:
Might be nice to factor this out into its own function, perhaps. Something like `get_tool_modules`?